src/recovery/impl/thread_local_logger.h

```diff
@@ -64,16 +67,8 @@ class ThreadLocalLogger final : public LoggerBase {
         : thread_id(ThreadIdCounter.fetch_add(1)),
           durable_epoch(EpochFramework::THREAD_OFFLINE),
           truncated_epoch(0),
-          log_file(GetLogFileName(), std::fstream::out | std::fstream::binary |
-                                         std::fstream::ate) {}
+          log_file(0){}
```

Comment:
I am hard to read what type of constructor is invoked from `log_file(0)`; I guessed that it might be opening the file `"0"`.

I recommend deleting this line.
Then the default constructor will be invoked implicitly, and thus the `log_file` will not be associated with any file.

ref: http://www.cplusplus.com/reference/fstream/fstream/fstream/

Reply:
it's not actually open anything. Just set undefined anything. When I drop the line here, the reference (in-flight file io open `*1`) can not detect the un-opened file. probably it is caused the implicit constructor creates an instance.

Change the value from `0` to `nullptr` is an option we can save the semantics here. (i confirmed `nullptr` works as well)
Or, any idea to get the info that the instance is not associated to any file, is welcome.

1: https://github.com/LineairDB/LineairDB/pull/266/files#diff-f9100455e6c7f178d07d9c55091dd60f6cfce2e625e60dd6b1fa897d93481be2R84

Reply:
[8f3c903](https://github.com/LineairDB/LineairDB/pull/266/commits/8f3c903189af474cab6bc31e8f1385b566a7c1bf)
merged.